src/i_winmusic.c

```diff
@@ -1690,36 +1701,21 @@ static void I_WIN_ShutdownMusic(void)
         return;
     }
 
-    I_WIN_StopSong(NULL);
-    I_WIN_UnRegisterSong(NULL);
-
-    // Reset device at shutdown.
-    buffer.position = 0;
-    ResetDevice();
-    StreamOut();
-    mmr = midiStreamRestart(hMidiStream);
-    if (mmr != MMSYSERR_NOERROR)
+    if (win_midi_state != STATE_STOPPED)
     {
-        MidiError(VB_WARNING, "midiStreamRestart", mmr);
-    }
-    WaitForSingleObject(hBufferReturnEvent, PLAYER_THREAD_WAIT_TIME);
-    mmr = midiStreamStop(hMidiStream);
-    if (mmr != MMSYSERR_NOERROR)
-    {
-        MidiError(VB_WARNING, "midiStreamStop", mmr);
+        I_WIN_StopSong(NULL);
+        I_WIN_UnRegisterSong(NULL);
     }
 
```

Comment:
Done in https://github.com/fabiangreffrath/woof/pull/1248/commits/62389bba5db1d960378b5ada3c9514db642cf221

Reply:
I've tried to fix this here: https://github.com/fabiangreffrath/woof/pull/1248/commits/84cdc6ad65f8668ea950245fb72a6b659ac5b8ca Now the code is very close to what we had before. I'm wondering if it will crash again (I still can't get crash anywhere).

> (1) implies midiStreamStop() but I don't think the buffer is immediately available after calling that function, so care must be taken (wait until dwFlags is MHDR_DONE ?).

I'm also concerned about `midiStreamStop()`, but ZMusic and others just call it without waiting. Perhaps that would work.